Merge back 'chore_release-8.5.0' into 'chore_release-pd-8.5.0'

ddcc4 opened the pull request:
This picks up the changes to `chore_release-8.5.0` from today.
Will merge at end-of-day.

andySigler committed:
fix(shared-data): Z noise during 96ch submerge/retract (glycerol only) (#18643)

mjhuff committed:
fix(app): queue notification refetch if mqtt topic update occurs while refetching (#18694)

Closes RABR-781

If MQTT informs the app that it should refetch a particular topic while the app is actively refetching that topic, the app doesn't perform an additional refetch after the first refetch completes. This creates a race potential condition.

To fix, the app always needs to refetch again if MQTT tells the app an update is available while the app is in the process of refetching.

jbleon95 committed:
feat(api): add keep_last_tip argument to liquid class transfer methods (#18693)

Adds a new argument `keep_last_tip` to three liquid class transfer functions (`transfer_with_liquid_class`,
`consolidate_with_liquid_class`, and `distribute_with_liquid_class`). If this is set to `True`, the last (or only tip for `never` and `once`) will not be dropped at the end of the high level transfer/consolidate/distribute. Otherwise, the last tip will be dropped, even if the tip policy is `never`.

sanni-t committed:
feat(api): do prepare for aspirate and airgap above well always (#18653)

Closes AUTH-1978
Follows #18585 

# Overview

When we aren't using LLD or estimating liquid heights based on well
volumes, we cannot verify if the retract position set in a liquid class
lies inside liquid. In such cases we run the risk of accidentally
aspirating liquid inside the tip (and eventually inside the pipette). So
this PR moves the pipette to a safe location above the well (2mm above
well top) before executing a `prepareForAspirate` or adding an air gap.

## Changelog

- added a check to compare the retract location and a safe location for
plunger movement (2mm above well top) and do a move to safe location if
needed
- added unit tests

## Risk assessment

Low. Supposed to reduce risk of accidental liquid aspiration.

mjhuff reviewed (approved):
App changes look good. Thank you!

jerader reviewed (approved):
🚀 probably should wait for Sanniti's [commit](https://github.com/Opentrons/opentrons/pull/18695) to be included before merging this in?

sanni-t committed:
feat(api): add new tip option of 'always' to consolidate and distribute (#18695)

Closes AUTH-1995

# Overview

Allows `new_tip="always"` as a valid tip option for LC-based consolidate
and distribute. This will make the new consolidate and distribute
functions behave same as PD:
- 'always' for consolidate will pick up a new tip before **the next
chunk** of aspirates. Or in other words, will pick up a new tip when
going back to aspirate *after* a dispense
- 'always' for distribute will pick up a new tip before every aspirate

*Note that this behavior is different from what 'always' does in the
old, non-LC-based consolidate and distribute*. The old functions would
pick up a tip only once in the beginning of the function even if the tip
option was 'always'.

## Risk assessment

Medium. Touches a part of existing code that determines tip pickups.

sanni-t committed:
fix(api): remove invalid errors for lc positioning check (#18698)

Replaces #18585

# Overview

**About the error removal**

Until now we have assumed, and pushed for the assumption, that a
submerge start point and retract end point should be above the aspirate
& dispense locations of the transfer properties specified in a liquid
class. This requirement was mainly added to avoid getting into
situations where the submerge start location or retract end location
could end up being inside the liquid, especially when we have no idea of
the liquid's height inside well.

The above situation was a concern because it could mean aspirating
liquid by mistake when trying to aspirate air gaps or move the plunger
to prepare for aspiration at either the submerge start point or retract
end point. But over the course, we have changed this behavior so that
`prepareForAspirate` and `addAirGap` both happen above the well, always
(#18653, #17800). So we no longer truly need to enforce the positioning
requirement of submerge start & retract end.

Additionally, consider this example-
An LC-based transfer that wants to set its dispense location as 5mm
below the liquid meniscus. Note that this is 5mm below where the
meniscus would be at the **end** of the dispense. Let's assume this
location is `Point(10, 10, 10)`.
Assuming the destination well is empty in the beginning, one could set
their submerge location to be `Point(10, 10, 8)` and that would be
totally fine since this location would actually not be inside liquid.
The tip should be allowed to start at `(10, 10, 8)` and then move up to
`(10, 10, 10)` to dispense. But the current check didn't allow that.
There's a couple of more examples like this where a submerge/ retract
point might lie below the aspirate/dispense points without causing any
real issues but our code won't allow it.

Because of these reasons we want to remove this check that looks for
order of these positions.
We will still be raising errors for when submerge start or retract end
positions lie inside the liquid.

**About log level change**

Internal users complained that the warning is too verbose and happens
way too often and it shows up in the default simulation output. So I'm
changing this to an `INFO` level so that it won't show up in the output
but will still be available in the logs as useful info.

## Risk assessment

None
